sys/kern/src/task.rs

```diff
@@ -846,13 +856,24 @@ pub fn force_fault(
             }
         }
     };
-    let supervisor_awoken =
-        tasks[0].post(NotificationSet(HUBRIS_FAULT_NOTIFICATION));
-    if supervisor_awoken {
-        NextTask::Specific(0)
-    } else {
-        NextTask::Other
-    }
+    notify_supervisor(tasks)
+}
+
+/// Puts a task into stopped state.
+///
+/// The task is designated by the `index` parameter. We need access to the
+/// entire task table, as well as the designated task, so that we can take the
+/// opportunity to notify the supervisor.
+///
+/// The task will not be scheduled again until requested to restart it again.
+///
+/// Returns a `NextTask` because the calling task cannot be scheduled anymore.
+pub fn exit_task(
+    tasks: &mut[Task],
+    index: usize,
+) -> NextTask {
+    tasks[index].set_healthy_state(SchedState::Stopped);
```

Comment:
Forgive my uninformed questions, but from what I can gather this function is called from user context. So the user context has direct access to tasks' state?